packages/edit-site/src/components/global-styles/border-panel.js

```diff
@@ -68,6 +65,52 @@ function useHasBorderWidthControl( name ) {
 	);
 }
 
+function useMultiOriginColors( name ) {
```

Comment:
There's some overlap between this hook and `useMultipleOriginColorsAndGradients` in `packages/block-editor/src/components/colors-gradients/use-multiple-origin-colors-and-gradients.js`. Do you think we should DRY that up? Or do you think it's too soon?

I can envision that a future developer might fix a bug or update a string in `useMultipleOriginColorsAndGradients` but not `useMultiOriginColors` because the Border panel that appears in the block settings sidebar is more obvious than the one that appears in global styles.

Reply:
> There's some overlap between this hook and useMultipleOriginColorsAndGradients

Thanks for the nudge to DRY this up. 

Upon revisiting this I've found the `useColorsPerOrigin` hook that was introduced to the site editor not too long before I opened this PR. I missed it as I was expecting a similar name to what was already in the block editor.

It took the same approach as the hook I previously had in this PR so I've updated these changes to leverage that instead. This means that all the color supports (background, link & text) along with border color use the same hook now.